File: app/src/main/java/com/msk/produtosperigosos/listas/ListaSubClasse.java
package com.msk.produtosperigosos.listas;

import android.annotation.SuppressLint;
import android.content.res.Resources;
import android.os.Bundle;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.MenuItem;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.msk.produtosperigosos.R;

public class ListaSubClasse extends AppCompatActivity {

    Resources r;
    // ELEMENTOS DA TELA
    private TextView classe, detalhe, numero, subclasse;
    private ListView subclasses;
    private View linhaView;
    private LayoutInflater inflaLista;
    private ImageView rotulo;
    // BUSCADOR DE INFORMACOES DA CLASSE
    private String[] nomeClasse, detClasse, nSubclasse, detSubclasse;
    private int nrClasse;
    private int[] rotuloId;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.sub_classes);
        r = getResources();

        Iniciador();
        usarActionBar();
        // BUSCA DADOS DA CLASSE DE RISCO
        Bundle envelope = getIntent().getExtras();
        nrClasse = envelope.getInt("nr");
        DefineInfoClasse(nrClasse);
        subclasses.setAdapter(new adapter());

    }

    private void Iniciador() {

        // ELEMENTO DA TELA
        subclasses = findViewById(R.id.lvSubclasse);
        // VETORES COM INFORMACOES DAS CLASSES
        nomeClasse = r.getStringArray(R.array.nr_classe_risco);
        detClasse = r.getStringArray(R.array.detalhe_classe);

    }

    private void DefineInfoClasse(int i) {

        if (i == 0) { // EXPLOSIVOS

            rotuloId = new int[]{R.drawable.rotulo_explosivo,
                    R.drawable.rotulo_explosivo, R.drawable.rotulo_explosivo,
                    R.drawable.rotulo_explosivo_4,
                    R.drawable.rotulo_explosivo_5,
                    R.drawable.rotulo_explosivo_6};
            nSubclasse = r.getStringArray(R.array.nr_sub_classe_1);
            detSubclasse = r.getStringArray(R.array.nome_sub_classe_1);
        }
        if (i == 1) { // GASES

            rotuloId = new int[]{R.drawable.rotulo_gas_inflamavel,
                    R.drawable.rotulo_gas_nao_toxico,
                    R.drawable.rotulo_gas_toxico};
            nSubclasse = r.getStringArray(R.array.nr_sub_classe_2);
            detSubclasse = r.getStringArray(R.array.nome_sub_classe_2);
        }
        if (i == 2) { // LIQUIDOS

            rotuloId = new int[]{R.drawable.rotulo_liquido_inflamavel};
            nSubclasse = new String[]{nomeClasse[i]};
            detSubclasse = new String[]{getResources().getString(
                    R.string.sem_subclasse)};
        }
        if (i == 3) { // SOLIDOS

            rotuloId = new int[]{R.drawable.rotulo_solido_inflamavel,
                    R.drawable.rotulo_combustao_expontanea,
                    R.drawable.rotulo_perigoso_quando_molhado};
            nSubclasse = r.getStringArray(R.array.nr_sub_classe_4);
            detSubclasse = r.getStringArray(R.array.nome_sub_classe_4);
        }
        if (i == 4) { // OXIDOS

            rotuloId = new int[]{R.drawable.rotulo_oxidante,
                    R.drawable.rotulo_peroxido_organico};
            nSubclasse = r.getStringArray(R.array.nr_sub_classe_5);
            detSubclasse = r.getStringArray(R.array.nome_sub_classe_5);
        }
        if (i == 5) { // TOXICOS

            rotuloId = new int[]{R.drawable.rotulo_toxico,
                    R.drawable.rotulo_nocivo,
                    R.drawable.rotulo_substancia_infectante};
            nSubclasse = r.getStringArray(R.array.nr_sub_classe_6);
            detSubclasse = r.getStringArray(R.array.nome_sub_classe_6);
        }
        if (i == 6) { // RADIOATIVO

            rotuloId = new int[]{R.drawable.rotulo_radioativo_veic, R.drawable.rotulo_fissil};
            nSubclasse = new String[]{nomeClasse[i], "Produtos considerados físseis"};
            detSubclasse = new String[]{getResources().getString(
                    R.string.sem_subclasse), "Utilizado para identificar produtos que podem apresentar fissão nuclear"};
        }
        if (i == 7) { // CORROSIVO

            rotuloId = new int[]{R.drawable.rotulo_corrosivo};
            nSubclasse = new String[]{nomeClasse[i]};
            detSubclasse = new String[]{getResources().getString(
                    R.string.sem_subclasse)};
        }
        if (i == 8) { // DIVERSOS

            rotuloId = new int[]{R.drawable.rotulo_substancias_diversas};
            nSubclasse = new String[]{nomeClasse[i]};
            detSubclasse = new String[]{getResources().getString(
                    R.string.sem_subclasse)};
        }

    }

    @SuppressLint("NewApi")
    private void usarActionBar() {

        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setHomeButtonEnabled(true);

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {

            case android.R.id.home:
                finish();
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    class adapter extends BaseAdapter {

        @Override
        public int getCount() {
            return detSubclasse.length + 3;
        }

        @Override
        public Object getItem(int arg0) {
            return null;
        }

        @Override
        public long getItemId(int position) {
            return position;
        }

        @Override
        public View getView(int nrLinha, View lView, ViewGroup parent) {

            linhaView = lView;
            inflaLista = getLayoutInflater();
            if (nrLinha == 0 || nrLinha == 2) {
                linhaView = inflaLista.inflate(R.layout.nome_classe, null);
                classe = linhaView
                        .findViewById(R.id.tvNomeClasse);
                if (nrLinha == 0)
                    classe.setText(nomeClasse[nrClasse]);
                else
                    classe.setText(r.getString(R.string.subclasses));
            } else if (nrLinha == 1) {
                linhaView = inflaLista.inflate(R.layout.detalhe_classe,
                        null);
                detalhe = linhaView
                        .findViewById(R.id.tvDetalheClasse);
                detalhe.setText(Html.fromHtml("<b>" + r.getString(R.string.dica_nome_classe) + "</b>" + "<br />" + detClasse[nrClasse]));
            } else {
                linhaView = inflaLista.inflate(
                        R.layout.item_subclasse_risco, null);
                rotulo = linhaView
                        .findViewById(R.id.ivRotuloSubClasse);
                numero = linhaView
                        .findViewById(R.id.tvNrSubclasse);
                subclasse = linhaView
                        .findViewById(R.id.tvNomeSubclasse);
                rotulo.setImageResource(rotuloId[nrLinha - 3]);
                numero.setText(nSubclasse[nrLinha - 3]);
                subclasse.setText(detSubclasse[nrLinha - 3]);
            }

            return linhaView;
        }

    }

}

File: src/main/java/com/msk/produtosperigosos/listas/ListaSubClasse.kt
package com.msk.produtosperigosos.listas

import android.annotation.SuppressLint
import android.content.res.Resources
import android.os.Bundle
import android.view.LayoutInflater
import android.view.MenuItem
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.ImageView
import android.widget.ListView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import com.google.android.material.color.MaterialColors
import com.msk.produtosperigosos.R

class ListaSubClasse : AppCompatActivity() {
    private var r: Resources? = null

    // ELEMENTOS DA TELA
    private var classe: TextView? = null
    private var detalhe: TextView? = null
    private var numero: TextView? = null
    private var subclasse: TextView? = null
    private var subclasses: ListView? = null
    private var inflaLista: LayoutInflater? = null
    private var rotulo: ImageView? = null

    // BUSCADOR DE INFORMACOES DA CLASSE
    private lateinit var nomeClasse: Array<String>
    private lateinit var detClasse: Array<String>
    private lateinit var nSubclasse: Array<String>
    private lateinit var detSubclasse: Array<String>
    private var nrClasse = 0
    private lateinit var rotuloId: IntArray

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.sub_classes)
        r = resources

        iniciador()
        usarActionBar()
        // BUSCA DADOS DA CLASSE DE RISCO
        val envelope = intent.extras
        nrClasse = envelope!!.getInt("nr")
        defineInfoClasse(nrClasse)
        subclasses!!.adapter = Adapter()
    }

    private fun iniciador() {
        // ELEMENTO DA TELA

        subclasses = findViewById(R.id.lvSubclasse)
        // VETORES COM INFORMACOES DAS CLASSES
        nomeClasse = r!!.getStringArray(R.array.nr_classe_risco)
        detClasse = r!!.getStringArray(R.array.detalhe_classe)
    }

    private fun defineInfoClasse(i: Int) {
        when (i) {
            0 -> { // EXPLOSIVOS
                rotuloId = intArrayOf(
                    R.drawable.rotulo_explosivo,
                    R.drawable.rotulo_explosivo, R.drawable.rotulo_explosivo,
                    R.drawable.rotulo_explosivo_4,
                    R.drawable.rotulo_explosivo_5,
                    R.drawable.rotulo_explosivo_6
                )
                nSubclasse = r!!.getStringArray(R.array.nr_sub_classe_1)
                detSubclasse = r!!.getStringArray(R.array.nome_sub_classe_1)
            }
            1 -> { // GASES
                rotuloId = intArrayOf(
                    R.drawable.rotulo_gas_inflamavel,
                    R.drawable.rotulo_gas_nao_toxico,
                    R.drawable.rotulo_gas_toxico
                )
                nSubclasse = r!!.getStringArray(R.array.nr_sub_classe_2)
                detSubclasse = r!!.getStringArray(R.array.nome_sub_classe_2)
            }
            2 -> { // LIQUIDOS
                rotuloId = intArrayOf(R.drawable.rotulo_liquido_inflamavel)
                nSubclasse = arrayOf(nomeClasse[i])
                detSubclasse = arrayOf(
                    resources.getString(
                        R.string.sem_subclasse
                    )
                )
            }
            3 -> { // SOLIDOS
                rotuloId = intArrayOf(
                    R.drawable.rotulo_solido_inflamavel,
                    R.drawable.rotulo_combustao_expontanea,
                    R.drawable.rotulo_perigoso_quando_molhado
                )
                nSubclasse = r!!.getStringArray(R.array.nr_sub_classe_4)
                detSubclasse = r!!.getStringArray(R.array.nome_sub_classe_4)
            }
            4 -> { // OXIDOS
                rotuloId = intArrayOf(
                    R.drawable.rotulo_oxidante,
                    R.drawable.rotulo_peroxido_organico
                )
                nSubclasse = r!!.getStringArray(R.array.nr_sub_classe_5)
                detSubclasse = r!!.getStringArray(R.array.nome_sub_classe_5)
            }
            5 -> { // TOXICOS
                rotuloId = intArrayOf(
                    R.drawable.rotulo_toxico,
                    R.drawable.rotulo_nocivo,
                    R.drawable.rotulo_substancia_infectante
                )
                nSubclasse = r!!.getStringArray(R.array.nr_sub_classe_6)
                detSubclasse = r!!.getStringArray(R.array.nome_sub_classe_6)
            }
            6 -> { // RADIOATIVO
                rotuloId = intArrayOf(R.drawable.rotulo_radioativo_veic, R.drawable.rotulo_fissil)
                nSubclasse = arrayOf(nomeClasse[i], "Produtos considerados físseis")
                detSubclasse = arrayOf(
                    resources.getString(
                        R.string.sem_subclasse
                    ), "Utilizado para identificar produtos que podem apresentar fissão nuclear"
                )
            }
            7 -> { // CORROSIVO
                rotuloId = intArrayOf(R.drawable.rotulo_corrosivo)
                nSubclasse = arrayOf(nomeClasse[i])
                detSubclasse = arrayOf(
                    resources.getString(
                        R.string.sem_subclasse
                    )
                )
            }
            8 -> { // DIVERSOS
                rotuloId = intArrayOf(R.drawable.rotulo_substancias_diversas)
                nSubclasse = arrayOf(nomeClasse[i])
                detSubclasse = arrayOf(
                    resources.getString(
                        R.string.sem_subclasse
                    )
                )
            }
        }
    }

    @SuppressLint("NewApi")
    private fun usarActionBar() {
        val toolbar = findViewById<Toolbar?>(R.id.toolbar)
        setSupportActionBar(toolbar)
        supportActionBar?.title = getString(R.string.dica_detalhe_classe)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeButtonEnabled(true)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    internal inner class Adapter : BaseAdapter() {
        override fun getCount(): Int {
            return if (detSubclasse.size == 1) 3 else detSubclasse.size + 3
        }

        override fun getItem(arg0: Int): Any? {
            return null
        }

        override fun getItemId(position: Int): Long {
            return position.toLong()
        }

        override fun getView(nrLinha: Int, convertView: View?, parent: ViewGroup): View {
            val view: View
            inflaLista = layoutInflater

            if (nrLinha == 0) {
                view = inflaLista!!.inflate(R.layout.nome_classe, parent, false)
                classe = view.findViewById(R.id.tvNomeClasse)
                classe?.text = nomeClasse[nrClasse]
            } else if (nrLinha == 1) {
                view = inflaLista!!.inflate(R.layout.detalhe_classe, parent, false)
                detalhe = view.findViewById(R.id.tvDetalheClasse)
                detalhe?.text = detClasse[nrClasse]
            } else if (nrLinha == 2 && detSubclasse.size > 1) {
                view = inflaLista!!.inflate(R.layout.nome_subclasse, parent, false)
                classe = view.findViewById(R.id.tvPossuiSubClasse)
            } else if (detSubclasse.size == 1) {
                view = inflaLista!!.inflate(R.layout.item_subclasse_risco, parent, false)
                rotulo = view.findViewById(R.id.ivRotuloSubClasse)
                numero = view.findViewById(R.id.tvNrSubclasse)
                subclasse = view.findViewById(R.id.tvNomeSubclasse)
                rotulo?.setImageResource(rotuloId[0])
                numero?.text = nSubclasse[0]
                subclasse?.text = detSubclasse[0]
            } else {
                view = inflaLista!!.inflate(R.layout.item_subclasse_risco, parent, false)
                rotulo = view.findViewById(R.id.ivRotuloSubClasse)
                numero = view.findViewById(R.id.tvNrSubclasse)
                subclasse = view.findViewById(R.id.tvNomeSubclasse)
                rotulo?.setImageResource(rotuloId[nrLinha - 3])
                numero?.text = nSubclasse[nrLinha - 3]
                subclasse?.text = detSubclasse[nrLinha - 3]
            }

            return view
        }
    }
}
